pkg/infoschema/issyncer/loader.go

```diff
@@ -289,6 +293,24 @@ func (l *Loader) LoadWithTS(startTS uint64, isSnapshot bool) (infoschema.InfoSch
 }
 
 func (l *Loader) skipLoadingDiff(diff *model.SchemaDiff) bool {
+	if l.loadDBFilter != nil {
+		// Always accept newly created schema as we cannot access its name in this context.
+		// Always accept `CREATE PLACEMENT POLICY`: its `schemaID` is ID of this poilcy but not zero.
+		if diff.Type == model.ActionCreateSchema || diff.Type == model.ActionCreatePlacementPolicy {
+			l.logger.Warn("Load DB filter was ignored.", zap.Int64("diff", diff.Version), zap.Stringer("type", diff.Type))
+			return false
+		}
+		// Always accept db unrelated DDLs.
+		if diff.SchemaID == 0 {
+			return false
+		}
+
+		is := l.infoCache.GetLatest()
+		schema, ok := is.SchemaByID(diff.SchemaID)
+		selected := ok && l.loadDBFilter(schema.Name)
+		return !selected
```

Comment:
since those logic are specific to BR, we better encapsulate all those into BR

we can abstract  interface like below, BR can have one specific implementation
```
type Filter interface {
    SkipLoadDiff(xxx params) bool
    SkipLoadSchema(xxx params) bool
    SkipMDLCheck(...) bool
}